vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -1,1011 +1,508 @@
-            if self.mor_objects_queue.contains(i_key) and self.mor_objects_queue.size(i_key, resource_type):
-                last = self.cache_config.get_last(CacheConfig.Morlist, i_key)
-                self.log.debug(
-                    "Skipping morlist collection: the objects queue for the "
-                    "resource type '%s' is still being processed "
-                    "(latest refresh was %ss ago)",
-                    ensure_unicode(resource_type),
-                    time.time() - last,
-                )
-                return
-
-        tags = ["vcenter_server:{}".format(ensure_unicode(instance.get('name')))]
-        regexes = {
-            'host_include': instance.get('host_include_only_regex'),
-            'vm_include': instance.get('vm_include_only_regex'),
-        }
-        include_only_marked = is_affirmative(instance.get('include_only_marked', False))
-
-        # Discover hosts and virtual machines
-        server_instance = self._get_server_instance(instance)
-        use_guest_hostname = is_affirmative(instance.get("use_guest_hostname", False))
-        all_objs = self._get_all_objs(
-            server_instance, regexes, include_only_marked, tags, use_guest_hostname=use_guest_hostname
+    def refresh_metrics_metadata_cache(self):
+        """Request the list of counters (metrics) from vSphere and store them in a cache."""
+        self.log.info(
+            "Refreshing the metrics metadata cache. Collecting all counters metadata for collection_level=%d",
+            self.collection_level,
         )
-
-        self.mor_objects_queue.fill(i_key, dict(all_objs))
-        self.cache_config.set_last(CacheConfig.Morlist, i_key, time.time())
-
-    @trace_method
-    def _process_mor_objects_queue_async(self, instance, mors):
-        """
-        Process a batch of items popped from the objects queue by querying the available
-        metrics for these MORs and then putting them in the Mor cache
-        """
-        t = time.time()
-        i_key = self._instance_key(instance)
-        server_instance = self._get_server_instance(instance)
-        perfManager = server_instance.content.perfManager
-
-        # For non realtime metrics, we need to specifically ask which counters are available for which entity,
-        # so we call perfManager.QueryAvailablePerfMetric for each cluster, datacenter, datastore
-        # This should be okay since the number of such entities shouldn't be excessively large
-        for mor in mors:
-            mor_name = str(mor['mor'])
-            available_metrics = {m.counterId for m in perfManager.QueryAvailablePerfMetric(entity=mor["mor"])}
-            try:
-                self.mor_cache.set_metrics(i_key, mor_name, self._compute_needed_metrics(instance, available_metrics))
-            except MorNotFoundError:
-                self.log.error("Object '%s' is missing from the cache, skipping. ", ensure_unicode(mor_name))
+        t0 = Timer()
```

Comment:
Any reason to use this rather than just track manually?

Reply:
You mean with something like `t0 = time.time(); #do_stuff; print(time.time()-t0)`?

No no strong reason, I reused what we have in the previous implementation but both can work. Want me to change it?